mapr/datadog_checks/mapr/mapr.py

```diff
@@ -0,0 +1,153 @@
+To support older versions of MapR, the check should be updated to subscribe those multiple topics.
+"""
+
+
+class MaprCheck(AgentCheck):
+    def __init__(self, name, init_config, instances):
+        super(MaprCheck, self).__init__(name, init_config, instances)
+        self._conn = None
+        self.hostname = self.instance.get('hostname', get_fqdn())
+        self.streams_count = self.instance.get('streams_count', 2)
+        self.topic_path = "{stream_path}/{stream_id}:{topic_name}".format(
+            stream_path=self.instance.get('stream_path', DEFAULT_STREAM_PATH),
+            stream_id=get_stream_id_for_topic(self.hostname, rng=self.streams_count),
+            topic_name=self.hostname,
+        )
+        self.allowed_metrics = [re.compile(w) for w in self.instance.get('metric_whitelist', [])]
+        self.base_tags = self.instance.get('tags', [])
+        self.is_first_check_run = True
+
+        auth_ticket = self.instance.get('ticket_location')
+        if auth_ticket:
+            os.environ['MAPR_TICKETFILE_LOCATION'] = auth_ticket
+        elif not os.environ.get('MAPR_TICKETFILE_LOCATION'):
+            self.log.info("MAPR_TICKETFILE_LOCATION environment variable not set, this may cause authentication issues")
+
+    def check(self, _):
+        if ck is None:
+            raise CheckException(
+                "confluent_kafka was not imported correctly, make sure the library is installed and that you've "
+                "set LD_LIBRARY_PATH correctly. Please refer to datadog documentation for more details."
+            )
+
+        try:
+            conn = self.get_connection()
+        except Exception:
+            self.service_check(
+                SERVICE_CHECK, AgentCheck.CRITICAL, self.base_tags + ['topic_path:{}'.format(self.topic_path)]
+            )
+            raise
+
+        submitted_metrics_count = 0
+
+        while True:
+            msg = conn.poll(timeout=0.5)
+            if msg is None:
+                # Timed out, no more messages
+                break
+
+            if msg.error() is None:
+                # Metric received
+                try:
+                    metric = json.loads(ensure_unicode(msg.value()))[0]
+                    metric_name = metric['metric']
+                    if self.should_collect_metric(metric_name):
+                        # Will sometimes submit the same metric multiple time, but because it's only
+                        # gauges and monotonic_counter that's fine.
+                        self.submit_metric(metric)
+                        submitted_metrics_count += 1
+                except Exception as e:
+                    self.log.warning("Received unexpected message %s, wont be processed", msg.value())
+                    self.log.exception(e)
+            elif msg.error().code() != ck.KafkaError._PARTITION_EOF:
+                # Real error happened
+                raise CheckException(msg.error())
```

Comment:
Could you add a doc about why it's ok to not raise error for `ck.KafkaError._PARTITION_EOF` ?